operator/src/main/java/oracle/kubernetes/operator/wlsconfig/WlsServerConfig.java

```diff
@@ -156,6 +157,52 @@ public boolean isAdminPortEnabled() {
     return adminPort != null;
   }
 
+  public String getAdminProtocolChannelName() {
+    String adminProtocolChannel = null;
+    if (networkAccessPoints != null) {
+      for (NetworkAccessPoint nap : networkAccessPoints) {
+        if (nap.isAdminProtocol()) {
+          adminProtocolChannel = LegalNames.toDNS1123LegalName(nap.getName());
+          break;
+        }
+      }
+    }
+    if (adminProtocolChannel == null) {
+      if (adminPort != null) {
+        adminProtocolChannel = "default-admin";
+      } else if (sslListenPort != null) {
+        adminProtocolChannel = "default-secure";
+      } else if (listenPort != null) {
+        adminProtocolChannel = "default";
+      }
+    }
+
```

Comment:
Cool: you're figuring out the 'true' admin port here.   I think one addition is needed - I suspect the admin protocol needs to be deduced too.   Default-admin and default-secure requires https instead of http.   I assume admin nap does too - but not sure.